packages/sanity/src/core/store/_legacy/history/createHistoryStore.ts

```diff
@@ -217,23 +209,25 @@ function restore(
     }),
     mergeMap((restoredDraft) => {
       if (options?.useServerDocumentActions) {
-        return vXClient.observable.request({
-          url: `/data/actions/${dataset}`,
-          method: 'post',
-          body: {
-            actions: [
-              {
-                actionType: 'sanity.action.document.replaceDraft',
-                publishedId: documentId,
-                attributes: restoredDraft,
-              },
-            ],
+        return client.observable.action([
+          {
```

Comment:
Including the "create" action seems to cause restoring to fail with "there is already a published document with that ID" (if the document has been published) or "there is already a draft document with that ID" (if it has not)

I have probably missed some conversations on its inclusion, so I don't feel comfortable removing it -- please let me know if I can help!

<img width="1161" alt="Screenshot 2024-05-28 at 4 41 36 PM" src="https://github.com/sanity-io/sanity/assets/3969996/bf4ef50b-43f4-4306-a906-5fdea28c005a">

Reply:
Great catch @cngonzalez! Looking into it.

Reply:
Fixed by 614eb98a4debbb7c8315ca5c4015ed688614380a